axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -380,6 +335,70 @@ public void disconnect() {
         return shutdownLatch.initiateShutdown();
     }
 
+    /**
+     * A {@link QueryHandler} implementation serving as a wrapper around the local {@link QueryBus} to push through the
+     * message handling and subscription query registration.
+     */
+    private static class LocalSegmentAdapter implements QueryHandler {
```

Comment:
The only thing that might simplify the code a littlebit, is creating an non-static inner class here. The class is purely meant for internal use and only a single instance exists.

Reply:
I was having that debate in my mind actually... 
Yes, this is cleaner. But, more verbose. 

Let's go with the no-static class this time.
The `LocalSegmentAdapter` isn't _that_ complicated that it isn't clear it's using variables from its parent class.
If it does become too large or complex, I would be in favour of switching it back too.